nodes/generators_extended/hexa_grid.py

```diff
@@ -0,0 +1,321 @@
+from sverchok.ui.sv_icons import custom_icon
+from sverchok.utils.geom import circle
+from sverchok.utils.sv_mesh_utils import mesh_join
+
+DEBUG = False
+
+gridTypeItems = [
+    ("RECTANGLE", "Rectangle", "", custom_icon("SV_HEXA_GRID_RECTANGLE"), 0),
+    ("TRIANGLE", "Triangle", "", custom_icon("SV_HEXA_GRID_TRIANGLE"), 1),
+    ("DIAMOND", "Diamond", "", custom_icon("SV_HEXA_GRID_DIAMOND"), 2),
+    ("HEXAGON", "Hexagon", "", custom_icon("SV_HEXA_GRID_HEXAGON"), 3)]
+
+
+def get_hexagon_grid(r, level, center):
+    dx = r * 3 / 2
+    dy = r * sqrt(3)
+
+    num = level - 1
+    numX = 2 * num + 1
+    numY = list(map(lambda x: 2 * num - abs(num - x) + 1, range(numX)))
+
+    cx = num * dx if center else 0
+    cy = num * dy / 2 if center else 0
+
+    grid_values = []
+    grid_x = 0
+    grid_y = 0
+
+    for i in range(numX):
+        grid_y = - (numY[i] - numY[0]) * dy / 2
+        for j in range(numY[i]):
+            grid_values.append((grid_x - cx, grid_y - cy, 0.0))
+            grid_y += dy
+
+        grid_x += dx
+
+    return grid_values
+
+
+def get_diamond_grid(r, level, center):
+    dx = r * 3 / 2
+    dy = r * sqrt(3)
+
+    num = level - 1
+    numX = 2 * num + 1
+    numY = list(map(lambda x: num - abs(num - x) + 1, range(numX)))
+
+    cx = num * dx if center else 0
+    cy = 0
+
+    grid_values = []
+    grid_x = 0
+    grid_y = 0
+
+    for i in range(numX):
+        grid_y = - (numY[i] - 1) * dy / 2
+        for j in range(numY[i]):
+            grid_values.append((grid_x - cx, grid_y - cy, 0.0))
+            grid_y += dy
+
+        grid_x += dx
+
+    return grid_values
+
+
+def get_triangle_grid(r, level, center):
+    dx = r * 3 / 2
+    dy = r * sqrt(3)
+
+    num = level
+
+    cx = (num - 1) * r if center else 0
+    cy = 0
+
+    grid_values = []
+    grid_x = 0
+    grid_y = 0
+
+    for i in range(num):
+        grid_y = -int((i + 1) / 2) * dy
+        grid_y += dy / 2 if (i % 2 != 0) else 0
+        for j in range(i + 1):
+            grid_values.append((grid_x - cx, grid_y - cy, 0.0))
+            grid_y += dy
+
+        grid_x += dx
+
+    return grid_values
+
+
+def get_rectangle_grid(r, numx, numy, center):
+    dx = r * 3 / 2
+    dy = r * sqrt(3)
+
+    cx = (numx - 1) * dx / 2 if center else 0
+    cy = (numy - 1.0 + 0.5 * (numx > 1)) * dy / 2 if center else 0
+
+    grid_values = []
+    grid_x = 0
+    grid_y = 0
+
+    for i in range(numx):
+        grid_y = dy / 2 if (i % 2 != 0) else 0
+        for j in range(numy):
+            grid_values.append((grid_x - cx, grid_y - cy, 0.0))
+            grid_y += dy
+
+        grid_x += dx
+
+    return grid_values
+
+
+def generate_tiles(radius, angle, join, gridList):
+    verts, edges, polys = circle(radius, radians(angle), 6, None, 'pydata')
+
+    vertList = []
+    edgeList = []
+    polyList = []
+    for grid in gridList:
+        for cx, cy, cz in grid:
+            verts2 = [(x + cx, y + cy, z + cz) for x, y, z in verts]
```

Comment:
Yeah, I stayed away from using vectors since the tiling is just a simple xy translation and I figured it would be faster to just do cx/cy addition.